### PR TITLE
Add manifest for org.kde.akregator

### PR DESCRIPTION
**Pull request by flyingcakes85**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

<hr>

### Notes:

I get a `kf.notifications` message on launching Akregator. I'm not sure if this is referring to a potential packaging issue.

```
kf.notifications: env says KDE is running but SNI unavailable -- check KDE_FULL_SESSION and XDG_CURRENT_DESKTOP
```

### CODE REVIEW

**Comment by flyingcakes85** [comment]
bot, build org.kde.akregator

File: gpgme.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+    "name": "gpgme",
```
Comment (hfiguiere) [inline gpgme.json:2]: this is in the runtime.

File: org.kde.akregator.json [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+                {
+                    "name": "openldap",
+                    "config-opts": [
+                        "--disable-static",
+                        "--disable-slapd"
+                    ],
+                    "sources": [
+                        {
+                            "type": "archive",
+                            "url": "http://mirror.eu.oneandone.net/software/openldap/openldap-release/openldap-2.4.44.tgz",
```
Comment (hfiguiere) [inline org.kde.akregator.json:277]:
```suggestion
                            "url": "https://mirror.eu.oneandone.net/software/openldap/openldap-release/openldap-2.4.44.tgz",
```

File: org.kde.akregator.json [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DBUILD_TESTING=OFF",
+                "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/steveire/grantlee/archive/refs/tags/v5.2.0.tar.gz",
+                    "dest-filename": "grantlee-5.2.0.tar.gz",
```
Comment (hfiguiere) [inline org.kde.akregator.json:135]:
```suggestion
```

you don't need this

File: org.kde.akregator.json [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--share=network",
+        "--share=ipc",
+        "--device=dri",
+        "--own-name=org.freedesktop.Akonadi",
+        "--own-name=org.freedesktop.Akonadi.*",
+        "--own-name=org.freedesktop.Akonadi.Control.*",
+        "--own-name=org.freedesktop.Akonadi.Agent.*",
+        "--own-name=org.freedesktop.Akonadi.Resource.*",
```
Comment (hfiguiere) [inline org.kde.akregator.json:19]: isn't the first one enough

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: gpgme.json (hunk 1 above)
Comment (hfiguiere) [reply gpgme.json:2]: Or at least the dependencies since it seems that gpgme is missing Qt support in the runtime.

Comment (flyingcakes85) [reply gpgme.json:2]: @hfiguiere 
gpgme seems to be required. Build fails without that.

File: org.kde.akregator.json (hunk 4 above)
Comment (flyingcakes85) [reply org.kde.akregator.json:19]: Last three indeed seem to be redundant. So, I'll have these two :
```
"--own-name=org.freedesktop.Akonadi",
"--own-name=org.freedesktop.Akonadi.*",
```

I'm supposing those two don't imply each other.

File: org.kde.akregator.json (hunk 2 above)
Comment (flyingcakes85) [reply org.kde.akregator.json:277]: Thanks for bringing to notice. I rechecked and found another http link. I've replaced both with https.

File: org.kde.akregator.json (hunk 3 above)
Comment (flyingcakes85) [reply org.kde.akregator.json:135]: Removed :+1:

**Comment by flyingcakes85** [comment]
bot, build org.kde.akregator

**Comment by flyingcakes85** [comment]
> ]2;flatpak-builder: Downloading libkleoDownloading https://download.kde.org/stable/release-service/21.12.1/src/libkleo-21.12.1.tar.xz
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Failed to download sources: module libkleo: Error in the HTTP2 framing layer

But I can access the link :confused:

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
bot, build org.kde.akregator

File: gpgme.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                }
+            ],
+            "cleanup": [
+                "/lib/libassuan.la"
+            ]
+        }
+    ],
+    "cleanup": [
+        "/lib/libgpgme*.la",
+        "/lib/libqgpgme.la"
```
[the remark this answers is not in the document]
Comment (flyingcakes85) [reply gpgme.json]: Fixed in new commit 
https://github.com/flathub/flathub/pull/2769/files#diff-fe7557f72be96f1cd463f08d9a88555291b77040aff27d4ab6053d0ce7e8fd39R59

File: gpgme.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
+{
+    "name": "gpgme",
+    "sources": [
+        {
+            "type": "archive",
+            "url": "https://gnupg.org/ftp/gcrypt/gpgme/gpgme-1.15.1.tar.bz2",
+            "sha256": "eebc3c1b27f1c8979896ff361ba9bb4778b508b2496c2fc10e3775a40b1de1ad"
+        }
+    ],
+    "modules": [
```
[the remark this answers is not in the document]
Comment (flyingcakes85) [reply gpgme.json]: Thanks for clearing it up. I've removed those modules.

File: org.kde.akregator.json [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+                    "url": "https://download.kde.org/stable/release-service/21.12.1/src/akonadi-mime-21.12.1.tar.xz",
+                    "sha256": "a6016f9c58fa899773c3f79f61aaae179a508e37405771a836bafcde72c22fca"
+                }
+            ]
+        },
+        {
+            "name": "grantlee",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DBUILD_TESTING=OFF",
```
[the remark this answers is not in the document]
Comment (flyingcakes85) [reply org.kde.akregator.json]: Removed in new commit 
https://github.com/flathub/flathub/pull/2769/commits/0eb4891f47872cf1cd26a634f645b19f99bc4dae#diff-fe7557f72be96f1cd463f08d9a88555291b77040aff27d4ab6053d0ce7e8fd39L125

**Comment by flyingcakes85** [comment]
Changes in last commit:

- Simplify cleanup args for gpgme (from review)
- Remove modules for gpgme as they are in runtime (from review)
- Remove testing flag from grantlee (from review)
- Put boost and gpgme in main manifest (since the external files were small now)
- Move kpimtexteditor above libkleo, as former is optional dependency of libkleo

**Comment by flyingcakes85** [comment]
bot, build org.kde.akregator

**Comment by flyingcakes85** [comment]
This time same previous error, but for akonadi
> flatpak-builder: Downloading akonadi-searchDownloading https://download.kde.org/stable/release-service/21.12.1/src/akonadi-search-21.12.1.tar.xz
 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Failed to download sources: module akonadi-search: Error in the HTTP2 framing layer

**Comment by hfiguiere** [comment]
I had the error locally and tried again and it worked.

bot, build org.kde.akregator

**Comment by flyingcakes85** [comment]
Failed already.

https://github.com/curl/curl/issues/3750

https://bugzilla.redhat.com/show_bug.cgi?id=1690971

I found these links. Though I'm not the one to be able to understand it.

**Comment by hfiguiere** [comment]
It a problem on KDE side I think.

File: org.kde.akregator.json [hunk 8]
```diff
@@ -0,0 +1,495 @@
+{
+    "id": "org.kde.akregator",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15-21.08",
+    "sdk": "org.kde.Sdk",
+    "base": "io.qt.qtwebengine.BaseApp",
+    "base-version": "5.15-21.08",
+    "command": "akregator",
```
Comment (hfiguiere) [inline org.kde.akregator.json:8]:
```suggestion
    "command": "akregator",
    "rename-icon": "akregator",
```

There were complaints in the build output, not sure how it passed.

File: org.kde.akregator.json [hunk 9]
```diff
@@ -0,0 +1,495 @@
...
+        "--talk-name=org.kde.kglobalaccel",
+        "--talk-name=org.kde.JobViewServer",
+        "--talk-name=org.kde.klauncher5",
+        "--talk-name=org.kde.knotify",
+        "--talk-name=org.kde.kwalletd5",
+        "--talk-name=org.kde.kcookiejar5",
+        "--system-talk-name=org.freedesktop.NetworkManager"
+    ],
+
+    "modules": [
```
Comment (hfiguiere) [inline org.kde.akregator.json:33]:
```suggestion
    "cleanup-commands": [
        "# Cleanup after QtWebEngine",
        "rm -rf /app/include",
        "rm -rf /app/lib/{cmake,mkspecs,pkgconfig} /app/lib/*/*.la"
    ],
    "modules": [
```

Cleanup after webengine

File: org.kde.akregator.json [hunk 10]
```diff
@@ -0,0 +1,495 @@
...
+        {
+            "name": "boost",
+            "buildsystem": "simple",
+            "build-commands": [
+                "./bootstrap.sh --prefix=/app --with-libraries=system",
+                "./b2 -j $FLATPAK_BUILDER_N_JOBS install"
+            ],
```
Comment (hfiguiere) [inline org.kde.akregator.json:40]:
```suggestion
            ],
            "cleanup": [
                "*.a"
            ],
```

File: org.kde.akregator.json [hunk 11]
```diff
@@ -0,0 +1,495 @@
...
+                {
+                    "name": "sqlite3",
+                    "config-opts": [
+                        "--enable-threadsafe",
+                        "--enable-threads-override-locks"
+                    ],
+                    "build-options": {
+                        "cflags": "-DSQLITE_ENABLE_UNLOCK_NOTIFY=1"
+                    },
+                    "sources": [
```
Comment (hfiguiere) [inline org.kde.akregator.json:104]:
```suggestion
                    "cleanup": [
                        "*.a"
                    ],
                    "sources": [
```

File: org.kde.akregator.json [hunk 12]
```diff
@@ -0,0 +1,495 @@
...
+                        {
+                            "type": "archive",
+                            "url": "https://mirror.eu.oneandone.net/software/openldap/openldap-release/openldap-2.4.44.tgz",
+                            "sha256": "d7de6bf3c67009c95525dde3a0212cc110d0a70b92af2af8e3ee800e81b88400"
+                        }
+                    ],
+                    "cleanup": [
+                        "/bin/ldap*",
+                        "/sbin"
```
Comment (hfiguiere) [inline org.kde.akregator.json:304]:
```suggestion
                        "/share/man",
                        "/sbin",
                        "*.la"
```

File: org.kde.akregator.json [hunk 13]
```diff
@@ -0,0 +1,495 @@
...
+                "--disable-documentation"
+            ],
+            "cleanup": [
+                "/share/doc/xapian-core",
+                "/bin/xapian-*",
+                "/bin/copydatabase",
+                "/bin/quest",
+                "/bin/simpleexpand",
+                "/bin/simpleindex",
+                "/bin/simplesearch"
```
Comment (hfiguiere) [inline org.kde.akregator.json:328]:
```suggestion
                "/bin/simplesearch",
                "*.la"
```

File: org.kde.akregator.json [hunk 14]
```diff
@@ -0,0 +1,495 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/kuserfeedback/kuserfeedback-1.0.0.tar.xz",
+                    "sha256": "5a2f53ebb4b99a280757ca32bd9b686a7764a726e7e4d8bafee33acbb44b9db7"
+                }
+            ]
+        },
+        {
+            "name": "akregator",
+            "buildsystem": "cmake-ninja",
```
Comment (hfiguiere) [inline org.kde.akregator.json:481]:
```suggestion
            "buildsystem": "cmake-ninja",
            "post-install": [
                "mv ${FLATPAK_DEST}/share/mime/packages/akonadi-mime.xml ${FLATPAK_DEST}/share/mime/packages/${FLATPAK_ID}.xml"
            ],
```

This export the MIME file properly

**Review by hfiguiere — CHANGES_REQUESTED** [review]
This changes fix some issues, and cleanup better.

**Comment by hfiguiere** [comment]
also the app is very verbose on the command line. 

Including a lot of 
```
kf.kio.slaves.http: Can't communicate with kded_kcookiejar!
```
(I don't run KDE/Plasma, so that could be why, in that case, ignore it)

**Comment by hfiguiere** [comment]
The build passed on x86_64 though.

**Comment by flyingcakes85** [comment]
Pushed new changes.

> also the app is very verbose on the command line.
> 
> Including a lot of
> 
> ```
> kf.kio.slaves.http: Can't communicate with kded_kcookiejar!
> ```
> 
> (I don't run KDE/Plasma, so that could be why, in that case, ignore it)

Yes, because of not having KDE. I'm on KDE and I don't see those messages.

File: org.kde.akregator.json (hunk 11 above)
Comment (flyingcakes85) [reply org.kde.akregator.json:114]: Added  below with other sqlite cleanup

**Comment by flyingcakes85** [comment]
bot, build org.kde.akregator

File: org.kde.akregator.json (hunk 11 above)
Comment (hfiguiere) [reply org.kde.akregator.json:114]: good call. I had fixed that change locally.

**Comment by barthalion** [comment]
bot, build org.kde.akregator

**Comment by travier** [comment]
bot, build org.kde.akregator

**Comment by travier** [comment]
CC @flathub/kde

**Comment by flyingcakes85** [comment]
> Or use a more recent tarball

I've updated OpenLDAP to 2.6.1

**Comment by flyingcakes85** [comment]
bot, build org.kde.akregator

**Comment by CarlSchwan** [comment]
The app is already packaged in https://github.com/flathub/org.kde.kontact/blob/master/org.kde.kontact.json

**Comment by flyingcakes85** [comment]
Kontact uses Akregator as a dependency, but I believe Akregator can be used standalone too. Am I getting it right?

**Comment by travier** [comment]
This is a case where we can either keep both independent (but synced) or have the Kontact Flatpak expose both applications. Both have downsides unfortunately.

**Comment by hfiguiere** [comment]
you can indeed expose more than one app. The prime examples are LibreOffice (each individual app is exposed) and DIspplayCAL (it comes with some other utilities).

But I think in that case it will be less discoverable.

**Comment by CarlSchwan** [comment]
> This is a case where we can either keep both independent (but synced) or have the Kontact Flatpak expose both applications. Both have downsides unfortunately.

What do you mean by independent but synced? The two instance don't share the same config so they will display different content.

Also most dependencies in this manifest are not needed: akregator doesn't use Akonadi and having the kontact shell for one app is useless.

**Comment by flyingcakes85** [comment]
Despite the dependency overload, I'd still vote for having akregator as separate app on Flathub. Unless we can have two listings for same manifest on Flathub (one for Kontact, one for Akregator; but both downloading Kontact)

**Comment by nedrichards** [comment]
Given that the KDE app update workflow is pretty automated and ostree deduplication exists the only real pain is in multiple configuration files etc for users. I'm happy either way.

**Comment by travier** [comment]
I +1 for having a separated app. If users want to share config with other Kontact subapps, they can use the Kontact app and not this one.

**Pull request closed without merging**